agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -243,7 +304,7 @@ def current_state
 
       if service_container.running?
         'running'
-      elsif service_container.restarting?
+      elsif restarting?
         'restarting'
```

Comment:
Unable to get this to report `state: 'restarting'` in my testing... this `current_state` call happens immediately after the `ensure_started`, and at that point the container is still likely to be `running`. 

For this state to be useful and show up in `kontena service ls`, perhaps the worker should also `sync_state_to_master` at the point when it schedules the restart backoff. Also, consider keeping the `restarting` state for the `@restart_counter_reset_timer` duration, and only have it return to `running` once that expires?